dash/_patch.py

```diff
@@ -0,0 +1,117 @@
+    """
+    Patch a callback output value
+
+    Act like a proxy of the output prop value on the frontend.
+
+    Supported prop types: Dictionaries and lists.
+    """
+
+    def __init__(self, location=None, parent=None):
+        if location is not None:
+            self._location = location
+        else:
+            # pylint: disable=consider-using-ternary
+            self._location = (parent and parent._location) or []
+        if parent is not None:
+            self._operations = parent._operations
+        else:
+            self._operations = []
+
+    def __getitem__(self, item):
+        return Patch(location=self._location + [item], parent=self)
+
+    def __getattr__(self, item):
+        if item == "tolist":
+            # to_json fix
+            raise AttributeError
+        if item == "_location":
+            return self._location
+        if item == "_operations":
+            return self._operations
+        return self.__getitem__(item)
+
+    def __setattr__(self, key, value):
+        if key in ("_location", "_operations"):
+            self.__dict__[key] = value
+        else:
+            self.__setitem__(key, value)
+
+    def __delattr__(self, item):
+        self.__delitem__(item)
+
+    def __setitem__(self, key, value):
+        if value is _noop:
+            # The += set themselves.
+            return
+        self._operations.append(
+            _operation(
+                "Assign",
+                self._location + [key],
+                value=value,
+            )
+        )
+
+    def __delitem__(self, key):
+        self._operations.append(_operation("Delete", self._location + [key]))
+
+    def __add__(self, other):
+        self._operations.append(_operation("Add", self._location, value=other))
+        return _noop
+
+    def __iadd__(self, other):
+        self._operations.append(_operation("Add", self._location, value=other))
+        return _noop
+
+    def __sub__(self, other):
+        self._operations.append(_operation("Sub", self._location, value=other))
+        return _noop
+
+    def __isub__(self, other):
+        self._operations.append(_operation("Sub", self._location, value=other))
+        return _noop
+
+    def __mul__(self, other):
+        self._operations.append(_operation("Mul", self._location, value=other))
+        return _noop
+
+    def __imul__(self, other):
+        self._operations.append(_operation("Mul", self._location, value=other))
+        return _noop
+
+    def __truediv__(self, other):
+        self._operations.append(_operation("Div", self._location, value=other))
+        return _noop
+
+    def __itruediv__(self, other):
+        self._operations.append(_operation("Div", self._location, value=other))
+        return _noop
+
+    def append(self, item):
+        self._operations.append(_operation("Append", self._location, value=item))
+
+    def prepend(self, item):
+        self._operations.append(_operation("Prepend", self._location, value=item))
```

Comment:
`prepend` isn't actually a built-in list method, though it's useful and intuitive so I'm happy to keep it. But let's add `insert(index, item)`, which *is* a built-in method.

Other built-in list methods I don't think we need to implement, but maybe we should stub them so if someone tries to use them they at least get a reasonable error message, and if we implement them later it won't be a breaking change with existing attribute access? The ones I might put in this bucket: `clear`, `pop`, `remove`, `reverse`, `sort`.

Reply:
I think `sort` could have interesting use cases, not really sure what kind api we need to customize the sorting tho.
`reverse` an easy one: https://ramdajs.com/docs/#reverse so let's add it too.

Reply:
That's true, `sort` would be cool! But yeah, it's going to need a flexible API to be useful, so let's leave that out for now, we can always add it later. Then we should definitely stub it so it won't be a breaking change for anyone accessing a `sort` item inside their prop, they'll need to use `p["sort"]` from the start.